lib/i18n_sonder/workers/sync_translations.rb

```diff
@@ -39,15 +39,54 @@ def sync(approved_translations_only:)
         end
       end
 
-      def process_translation_result(translation_result, language, successful_syncs)
-        translations_by_model_and_id = translation_result.success
+      def process_translation_result(translation_result,
+                                     language,
+                                     successful_syncs,
+                                     handle_duplicates = false)
+        translations_and_source_text = translation_result.success
```

Comment:
Just to confirm I'm understanding this right, translations_and_source_text will have two different shapes depending on if the caller is using the old/status quo model_name keys or the newer one which has `source_text` and `translation` keys. Right now we assume that the callers will always pass in the right value for handle duplicates and that will trigger the logic to read from source_text and translation. This library seems like it won't be modified too much but wdyt about something like the `all_duplicate_translation` method is called if the `translations_and_source_text` contains both `source_text` and `translation` keys?

Reply:
interesting thought. i'll probably go one step further and separate the 2 use cases out into separate methods, just to be extra cautious.